tests/unit/test_download_manager.py

```diff
@@ -137,71 +137,4 @@ def test_find_next_file_index_one_scheduled_next_row_8000(self, mock_submit):
-
-    @patch("databricks.sql.cloudfetch.downloader.ResultSetDownloadHandler.is_file_download_successful",
-           return_value=False)
-    def test_check_if_download_successful_link_expired(self, mock_is_file_download_successful):
-        manager = self.create_download_manager()
-        handler = downloader.ResultSetDownloadHandler(manager.downloadable_result_settings, self.create_result_link())
-        handler.is_link_expired = True
-
-        status = manager._check_if_download_successful(handler)
-        mock_is_file_download_successful.assert_called()
-        assert not status
-        assert manager.fetch_need_retry
-
-    @patch("databricks.sql.cloudfetch.downloader.ResultSetDownloadHandler.is_file_download_successful",
-           return_value=False)
-    def test_check_if_download_successful_download_timed_out_no_retries(self, mock_is_file_download_successful):
-        manager = self.create_download_manager()
-        handler = downloader.ResultSetDownloadHandler(manager.downloadable_result_settings, self.create_result_link())
-        handler.is_download_timedout = True
-
-        status = manager._check_if_download_successful(handler)
-        mock_is_file_download_successful.assert_called()
-        assert not status
-        assert manager.fetch_need_retry
-
-    @patch("concurrent.futures.ThreadPoolExecutor.submit")
-    @patch("databricks.sql.cloudfetch.downloader.ResultSetDownloadHandler.is_file_download_successful",
-           return_value=False)
-    def test_check_if_download_successful_download_timed_out_1_retry(self, mock_is_file_download_successful, mock_submit):
-        manager = self.create_download_manager()
-        manager.downloadable_result_settings = download_manager.DownloadableResultSettings(
-            is_lz4_compressed=True,
-            download_timeout=0,
-            max_consecutive_file_download_retries=1,
-        )
-        handler = downloader.ResultSetDownloadHandler(manager.downloadable_result_settings, self.create_result_link())
-        handler.is_download_timedout = True
-
-        status = manager._check_if_download_successful(handler)
-        assert mock_is_file_download_successful.call_count == 2
-        assert mock_submit.call_count == 1
-        assert not status
-        assert manager.fetch_need_retry
-
-    @patch("databricks.sql.cloudfetch.downloader.ResultSetDownloadHandler.is_file_download_successful",
-           return_value=False)
-    def test_check_if_download_successful_other_reason(self, mock_is_file_download_successful):
-        manager = self.create_download_manager()
-        handler = downloader.ResultSetDownloadHandler(manager.downloadable_result_settings, self.create_result_link())
-
-        status = manager._check_if_download_successful(handler)
-        mock_is_file_download_successful.assert_called()
-        assert not status
-        assert manager.fetch_need_retry
+        assert manager._find_next_file_index(8000) is None
```

Comment:
By deleting the old tests and changing the logic, I think we missed retry coverage?

Reply:
I removed the old retry logic. No point in testing it now.